pdr_backend/pdr_dashboard/dash_components/plots.py

```diff
@@ -345,38 +345,63 @@ def process_payouts(
     """
     processed = ProcessedPayouts()
 
-    p = payouts.copy()
+    p = payouts.clone()
     processed.predictions = len(payouts)
-    p["predictions_crt"] = range(1, len(payouts) + 1)
+
+    p = p.with_columns(
+        pl.lit(1).alias("predictions_crt"),
+        pl.struct(["payout", "stake"])
+        .map_elements(lambda x: max(x["payout"], 0) - x["stake"], pl.Float64)
+        .alias("profit_change"),
+    )
+    p = p.with_columns(
+        pl.col("predictions_crt").cum_sum().alias("predictions_crt"),
+        pl.when(pl.col("payout") > 0).then(1).otherwise(0).alias("correct_prediction"),
+        pl.when(pl.col("payout") > 0)
+        .then(1)
+        .otherwise(0)
+        .cum_sum()
+        .alias("correct_predictions_crt"),
+        pl.col("slot")
+        .map_elements(lambda x: UnixTimeS(int(x)).to_milliseconds(), pl.Int64)
+        .alias("slot_in_unixts"),
+    )
```

Comment:
If I understood it correct, we can do it like that 

```
p = p.with_columns(
    [
        pl.lit(1).alias("predictions_crt"),
        (pl.col("payout").clip_min(0) - pl.col("stake")).alias("profit_change"),
        pl.when(pl.col("payout") > 0)
        .then(1)
        .otherwise(0)
        .alias("correct_prediction"),
        pl.col("slot")
        .cast(pl.Int64)
        .map_elements(lambda x: UnixTimeS(x).to_milliseconds(), pl.Int64).alias("slot_in_unixts")
    ]
)

p = p.with_columns(
    [
        pl.col("predictions_crt").cum_sum().alias("predictions_crt"),
        pl.col("correct_prediction").cum_sum().alias("correct_predictions_crt")
    ]
)
```